feat(PE-1564): use sfcc events api in react app

luqven opened the pull request:
This PR uses the SFCC events API inside the FE react application. It mimmics the use of the events API from #62. Using the events API in these index files allows us to define it with specific use cases for each application. This brings in events API usage from #62.



<!---GHSTACKOPEN-->
### Stacked PR Chain: PE-1564
| PR | Title | Status |  Merges Into  |
|:--:|:------|:-------|:-------------:|
|#61|build(PE-1564): split index file by application|**Approved**|-|
|#66|👉 feat(PE-1564): use sfcc events api in react app|**Approved**|#61|
|#67|~~fix(PE-1564): breakout css opacity bug~~|**Merged**|#66|
|#68|~~feat(PE-1564): feed imgix api creds to breakout component~~|**Merged**|#67|

<!---GHSTACKCLOSE-->




## Video 📹

https://user-images.githubusercontent.com/16711614/140834400-d4e94250-8a6e-41e8-a169-48a27756a52d.mov

frederickfogerty commented on cartridges/int_imgix_pd/cartridge/templates/default/imgix/imgixImage.isml:
```diff
@@ -0,0 +1,4 @@
+<!--- TEMPLATENAME: customBanner.isml --->
```
Can we please back out this change to using `imgixImage.isml` and revert to using the other template? Or copy the contents of the other template here? Thanks 🙌

frederickfogerty commented on frontend/src/components/SidebarApp.tsx:
```diff
@@ -4,10 +4,7 @@ import Imgix from "react-imgix";
-  const handleClick = () => {
-    console.log("clicked 'Add Image' button");
-  };
+export function App({ handleBreakoutOpen }: any) {
```
Let's fix this `any`

frederickfogerty commented on frontend/src/index-breakout.tsx:
```diff
@@ -3,39 +3,32 @@ import ReactDOM from "react-dom";
-  // reportWebVitals();
+    ReactDOM.render(
+      <React.StrictMode>
+        <App handleSubmit={handleSelect} />
```
Nice 👏

frederickfogerty commented on frontend/src/tests/services/imgixAPIService.test.ts:
```diff
@@ -69,7 +69,7 @@ test("should return assets in date modified descending order", async () => {
   const requestedUrl = mockRequest.mock.calls[0][0];
   const requestedURI = new URI(requestedUrl);
-  expect(requestedURI.getQueryParamValue("sort")).toBe("-date_modified");
+  expect(requestedURI.getQueryParamValue("sort")).toBe("date_created");
```
I think you want `-date_created` here, since that will sort the results so the most recently created files are first (AFAIK)

luqven replied:
`-date_created` has a weird edge case where if the timestamps are exactly the same between a number of assets then the sort will happen randomly, making pagination unpredictable.

frederickfogerty replied:
What's the best path forward here then? Out of interest, what does IM on the web use? I personally think that going forward with sorting by any sort of ascending date/time is a non-starter, since it will be frustrating for most people I imagine.

frederickfogerty commented on cartridges/int_imgix_pd/cartridge/experience/components/imgix/imageComponent.json:
```diff
@@ -4,21 +4,24 @@
         {
-          "id": "image_url",
-          "name": "Target Image",
+          "id": "image_sel",
```
Let's revert this line, since it is incompatible with `cartridges/int_imgix_pd/cartridge/experience/components/imgix/imageComponent.js`

frederickfogerty commented on cartridges/int_imgix_pd/cartridge/experience/components/imgix/imageComponent.json:
```diff
@@ -4,21 +4,24 @@
-          "id": "alt",
-          "name": "Alt Text",
-          "description": "The image alt text shown by the component.",
+          "id": "altText",
```
Let's also revert these lines

frederickfogerty reviewed (changes requested):
Looks good overal, just need to fix up a few things before merging

luqven commented:
@frederickfogerty [reverted](https://github.com/imgix/sf-commerce-cloud/pull/66/commits/546f27cce85f88b1ba957767a0685478b68f529b) to using `imageComponent.isml` and addressed some of your comments.

luqven commented:
Removed commit that checked in "working" react bundles. Will check that back in at the top of the stack where we can actually read from API keys.

frederickfogerty commented on frontend/src/index-sidebar.tsx:
```diff
@@ -3,39 +3,83 @@ import ReactDOM from "react-dom";
+      isRequired,
+      dataLocale,
+      displayLocale,
+    }: any) => {
```
Let's fix up this `any` before merging

frederickfogerty commented:
> Removed commit that checked in "working" react bundles. Will check that back in at the top of the stack where we can actually read from API keys.

What do you think about actually just using .gitignore for these files and instead only creating them locally, kind of like `dist` files in other libraries?

frederickfogerty reviewed (approved):
There's two things that I think need to be fixed here, but I think a higher priority is merging these PRs down to unblock development. Let's address these concerns in follow-up PRs
